src/deepsparse/pipeline.py

```diff
@@ -167,11 +175,31 @@ def __init__(
         self.onnx_file_path = self.setup_onnx_file_path()
         self.engine = self._initialize_engine()
 
-    def __call__(self, threadpool=None, *args, **kwargs) -> Union[BaseModel, Future]:
-        threadpool = threadpool or self._threadpool
-        if threadpool:
-            return threadpool.submit(self._run, *args, **kwargs)
-        return self._run(*args, **kwargs)
+    def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
+        executor = kwargs.get("executor")
+        if executor is not None:
+            kwargs.pop("executor")
+        else:
+            executor = self.executor
+
+        if self.in_dynamic_batch_mode():
+            # Blocking call in Dynamic Batch Mode
+            return executor.submit(self._run, *args, **kwargs).result()
```

Comment:
why are we submitting _run here if we will later use `self._run_with_dynamic_batch` to submit individual batches to the executor?